src/orion/benchmark/task/utils.py

```diff
@@ -0,0 +1,329 @@
+import dataclasses
+import itertools
+from functools import singledispatch
+from inspect import isclass
+from logging import getLogger as get_logger
+from typing import Any, Dict, Hashable, Iterable, Optional, Tuple, Type, TypeVar, Union
+
+import numpy as np
+
+# from warmstart.translation import SimpleTranslator, Translator
+# from .utils import dict_intersection, dict_union, field_dict
+from simple_parsing import Serializable
+from simple_parsing.helpers.hparams.hyperparameters import HyperParameters
+from simple_parsing.helpers.hparams.priors import (
+    LogUniformPrior,
+    Prior,
+    UniformPrior,
+)
+
+logger = get_logger(__name__)
+HP = TypeVar("HP")
+
+
+K = TypeVar("K")
+V = TypeVar("V")
+
+
+def zip_dicts(*dicts: Dict[K, V]) -> Iterable[Tuple[K, Tuple[Optional[V], ...]]]:
+    # If any attributes are common to both the Experiment and the State,
+    # copy them over to the Experiment.
+    keys = set(itertools.chain(*dicts))
+    for key in keys:
+        yield (key, tuple(d.get(key) for d in dicts))
+
+
+def dict_union(*dicts: Dict) -> Dict:
+    """ Simple dict union until we use python 3.9
+    
+    >>> from collections import OrderedDict
+    >>> a = OrderedDict(a=1, b=2, c=3)
+    >>> b = OrderedDict(c=5, d=6, e=7)
+    >>> dict_union(a, b)
+    OrderedDict([('a', 1), ('b', 2), ('c', 5), ('d', 6), ('e', 7)])
+    """
+    result: Dict = None  # type: ignore
+    for d in dicts:
+        if result is None:
+            result = type(d)()
+        result.update(d)
+    assert result is not None
+    return result
+
+
+def dict_intersection(*dicts: Dict[K, V]) -> Iterable[Tuple[K, Tuple[V, ...]]]:
+    common_keys = set(dicts[0])
+    for d in dicts:
+        common_keys.intersection_update(d)
+    for key in common_keys:
+        yield (key, tuple(d[key] for d in dicts))
+
+
+def intersection_over_union(a: Iterable[Hashable], b: Iterable[Hashable]) -> float:
+    a = set(a)
+    b = set(b)
+    return len(a.intersection(b)) / len(a.union(b))
+
+
+@singledispatch
```

Comment:
The next functions should probably be moved to a module for similarity metrics. But again, this could be moved out for another PR.